From e0eb62c0981f1aec80a6e434b2b737eeaeb09a7a Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Fri, 30 Aug 2013 15:18:34 +0800
Subject: [PATCH 0517/1072] ENGR00277458-1 ASoC: fsl: Use correct flag for
 imx_pcm_dma_init()

git://git.freescale.com/imx/linux-2.6-imx.git imx_3.10.17_1.0.0_beta
commit 36f3641146af3af2369f7b6f5941a6518b4700c7

We are using generic dma binding from devicetree for spdif, thus use
the correct flag.

Acked-by: Wang Shengjiu <b02247@freescale.com>
Signed-off-by: Nicolin Chen <b42378@freescale.com>
Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 sound/soc/fsl/fsl_spdif.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/fsl_spdif.c b/sound/soc/fsl/fsl_spdif.c
index db354dc8..c63e965 100644
--- a/sound/soc/fsl/fsl_spdif.c
+++ b/sound/soc/fsl/fsl_spdif.c
@@ -1187,9 +1187,7 @@ static int fsl_spdif_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	ret = imx_pcm_dma_init(pdev,  SND_DMAENGINE_PCM_FLAG_NO_RESIDUE |
-				     SND_DMAENGINE_PCM_FLAG_NO_DT |
-				     SND_DMAENGINE_PCM_FLAG_COMPAT);
+	ret = imx_pcm_dma_init(pdev, SND_DMAENGINE_PCM_FLAG_NO_RESIDUE);
 	if (ret) {
 		dev_err(&pdev->dev, "imx_pcm_dma_init failed: %d\n", ret);
 		goto error_component;
-- 
1.7.5.4

